[MDB Ignore] Unit Tests for Invalid Space Turfs (Area Bullshit Edition)

san7890 opened the pull request:
## About The Pull Request

So, there's some bullshit with the map loader(?) sometimes where it'll let space turfs spawn in spots where we REALLY don't want space turfs. Or, it could also just be a mapper screwing up. Anyways, we might miss these, so let's set up a broad Unit Test that checks and verifies that these round-ruining snagglers do _not_ exist.

In order to help me to do this, I standardized and fixed the nomenclature such that `/area/ruin/space` is default for any map file in `_maps/RandomRuins/SpaceRuins`, as well as it's subtypes. I also touched up how we handle shuttle areas in these scenarios. This got a lot of Unit Test noise filtered out, and is crucial for its functioning. It should also be how we did it from the start anyways. I added in an UpdatePaths for any compatible change, but it was completely non-workable for some of the area type updates.

I also fixed any organic bugs that didn't require an areas type update. Cool.

Placing space turfs on IceBox:
![image](https://user-images.githubusercontent.com/34697715/199177940-21c64964-1808-41b0-9a92-bf5b82eee2fa.png)

Organically found issues:
![image](https://user-images.githubusercontent.com/34697715/199177972-b27a89de-0e1a-41e5-8fa4-3bee1763b9da.png)

I also added a `planetary` variable to `/datum/map_config` because I didn't like the hack I was using to see if we had a planetary map, and I'd rather it just be an explicit variable in the map's JSON.

## Why It's Good For The Game

The less times we get Space Turfs showing up on IceBoxStation, the better. It also standardizes areas a bit more, which I like (we were using some incorrect ones in the wrong spots, so those were touched up in this PR as well). Like, if it's a space ruin, we don't need to use the lengthy `/area/ruin/unpowered/no_grav` when `/area/ruin/space` does the same thing.
## Changelog
Nothing in here should concern a player (unless I broke something)

Expect a few commits as I spam unit tests a few times and play whack-a-mole with bugs.

san7890 committed:
Unit Tests for Invalid Space Turfs (Area Bullshit Edition)

Brevity is the soul of wit.

san7890 committed:
UpdatePaths and Affected Maps

Also adds DM-side support for the UpdatePaths.

san7890 committed:
this was completely useless actually

san7890 committed:
i went too gloopy on updatepaths

yeah that was just silly

san7890 committed:
AREACOORD actually has the area name

 no need to be redundant

san7890 committed:
i don't like that it said lavaland for that error

that's a fix for a different day since it's out of scope here

also fixes the AI sat issues. MMM added new functionality to fix that, but didn't add it to his own map. very sad!

Mothblocks commented on code/modules/unit_tests/mapping.dm:
```diff
@@ -17,3 +17,61 @@
+	))
+
+	// We aren't planetary, so let's check area placements and ensure stuff lines up.
+	for(var/turf/iterated_turf in world)
```
This is really slow and so it should be in its own priority after all the faster tests.

@LemonInTheDark needs to do the same thing with one of his PRs. Work together to come up with a name and value so that you don't conflict with each other when I merge on or the other.

san7890 replied:
Priority assigned

san7890 replied:
Should be worth mentioning that according to runners this test takes roughly 0.6 seconds, while stuff like `mob_faction` (3.5 seconds), `preferences_implement_everything` (5.4 seconds), and `screenshot_humanoids` (31.3 seconds). I have no issue with giving it a priority if you need, but I'm just wondering where we want to assign all of the "funny define numbers" if these (and some more that I'm missing) take a magnitude longer to run.

The values are just from one test case, https://github.com/tgstation/tgstation/actions/runs/3367020406/jobs/5584114352, but those values seem accurate cross-comparing to similar CI runs.

LemonInTheDark replied:
I just rename the monkey number and used it
Mine is iterating all turfs in the world tho, so it should be a bit more then 0.6 seconds

san7890 committed:
Tram Fixes, Priority Assignment

today on mat-lab: What the fuck _was_ the dog doing?

san7890 committed:
small doc update

worth mentioning

san7890 committed:
actually i'm just going to code it in

i hated that comment

san7890 committed:
I may be stupid

(imagine i put nothing but a period here)

i left cruft in from an earlier iteration where i had a nested for loop within a for loop where it was completely un-necessary.

san7890 committed:
uses test_longer API

i'll merge master after this

san7890 committed:
Merge remote-tracking branch 'upstream/master' into i-fucking-hate-spacwe

san7890 committed:
Merge remote-tracking branch 'upstream/master' into i-fucking-hate-spacwe

san7890 committed:
Merge remote-tracking branch 'upstream/master' into i-fucking-hate-spacwe

san7890 committed:
Merge remote-tracking branch 'upstream/master' into i-fucking-hate-spacwe

san7890 commented:
![image](https://user-images.githubusercontent.com/34697715/202612803-4d0785bf-a1ac-40f8-aa1e-c635fa4b498a.png)

san7890 committed:
Merge remote-tracking branch 'upstream/master' into i-fucking-hate-spacwe

Mothblocks commented on code/modules/unit_tests/mapping.dm:
```diff
@@ -17,3 +17,61 @@
+
+/// Verifies that there are no space turfs inside a station area, or on any planetary z-level. Sometimes, these are introduced during the load of the map and are not present in the DMM itself.
+/// Let's just make sure that we have a stop-gap measure in place to catch these if they pop up so we don't put it onto production servers should something errant come up.
+/datum/unit_test/mapload_space_verification
```
We should probably make these different files, "mapping" is so broad

Mothblocks commented on code/modules/unit_tests/mapping.dm:
```diff
@@ -17,3 +17,61 @@
+	))
+
+	// We aren't planetary, so let's check area placements and ensure stuff lines up.
+	for(var/turf/iterated_turf in world)
```
@LemonInTheDark Should this be `ALL_TURFS()`?

san7890 replied:
using ALL_TURFS seems to make it fail alright:

```txt
[2022-11-21 01:31:24.967] Starting up round ID .
 - -------------------------
[2022-11-21 01:32:59.306] 	FAILURE #1: Space turf found in non-allowed area (/area/station/maintenance/tram/mid) at Central Tram Underpass (109,118,2)! Please ensure that all space turfs are in an /area/space! at code/modules/unit_tests/mapload_space_verification.dm:38
 - 	FAILURE #2: Space turf found in non-allowed area (/area/station/maintenance/tram/mid) at Central Tram Underpass (110,118,2)! Please ensure that all space turfs are in an /area/space! at code/modules/unit_tests/mapload_space_verification.dm:38
 - 	FAILURE #3: Space turf found in non-allowed area (/area/station/maintenance/tram/mid) at Central Tram Underpass (109,119,2)! Please ensure that all space turfs are in an /area/space! at code/modules/unit_tests/mapload_space_verification.dm:38
 - 	FAILURE #4: Space turf found in non-allowed area (/area/station/maintenance/tram/mid) at Central Tram Underpass (110,119,2)! Please ensure that all space turfs are in an /area/space! at code/modules/unit_tests/mapload_space_verification.dm:38
 - 	FAILURE #5: Space turf found in non-allowed area (/area/station/maintenance/tram/mid) at Central Tram Underpass (109,120,2)! Please ensure that all space turfs are in an /area/space! at code/modules/unit_tests/mapload_space_verification.dm:38
 - 	FAILURE #6: Space turf found in non-allowed area (/area/station/maintenance/tram/mid) at Central Tram Underpass (110,120,2)! Please ensure that all space turfs are in an /area/space! at code/modules/unit_tests/mapload_space_verification.dm:38
```

Mothblocks commented on code/modules/unit_tests/mapping.dm:
```diff
@@ -17,3 +17,61 @@
+	for(var/level in testable_levels)
+		var/testable_turfs = Z_TURFS(level)
+		for(var/turf/iterated_turf in testable_turfs)
+			if(isspaceturf(iterated_turf))
```
Loop over `turf/open/space` in the loop itself instead, that's all isspaceturf is.

san7890 committed:
Addresses Review

- Splits it out into a new file
- Removes the isspaceturf check
- tries out ALL_TURFS()

Mothblocks commented:
Conflicts

san7890 committed:
Merge remote-tracking branch 'upstream/master' into i-fucking-hate-spacwe

san7890 commented:
Addressed
